### PR TITLE
CTCP-4558: Create RemovePackageTypeYesNo page

### PR DESCRIPTION
**Pull request by justega247**
Screenshot 2024-03-22 at 13 16 21
Screenshot 2024-03-22 at 13 16 30
Screenshot 2024-03-22 at 13 16 41

### CODE REVIEW

File: app/controllers/houseConsignment/index/items/packages/RemovePackageTypeYesNoController.scala [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  private def packageType(implicit request: Request): PackageType = request.arg._2
+
+  private def form(packageType: PackageType): Form[Boolean] =
+    formProvider("houseConsignment.index.items.packages.removePackageTypeYesNo", packageType.toString)
+
+  def onPageLoad(arrivalId: ArrivalId, houseConsignmentIndex: Index, itemIndex: Index, packageIndex: Index, mode: Mode): Action[AnyContent] =
+    actions
+      .getStatus(arrivalId)
+      .andThen(getMandatoryPage.getFirst(NumberOfPackagesPage(houseConsignmentIndex, itemIndex, packageIndex)))
+      .andThen(getMandatoryPage.getSecond(PackageTypePage(houseConsignmentIndex, itemIndex, packageIndex))) {
```
Comment (jackheslop96) [inline app/controllers/houseConsignment/index/items/packages/RemovePackageTypeYesNoController.scala:64]: What if I add a new one, answer the first question, click back to the add-another page and click the remove link?

File: conf/messages.en [hunk 2]
```diff
@@ -941,4 +941,8 @@ houseConsignment.index.items.additionalReference.addAdditionalReferenceNumberYes
 
 houseConsignment.index.items.packages.addPackageShippingMarkYesNo.title = Do you want to add a shipping mark?
 houseConsignment.index.items.packages.addPackageShippingMarkYesNo.heading = Do you want to add a shipping mark?
-houseConsignment.index.items.packages.addPackageShippingMarkYesNo.error.required = Select yes if you want to add a shipping mark
\ No newline at end of file
+houseConsignment.index.items.packages.addPackageShippingMarkYesNo.error.required = Select yes if you want to add a shipping mark
+
+houseConsignment.index.items.packages.removePackageTypeYesNo.title = Are you sure you want to remove this package from item {1} in house consignment {0}?
+houseConsignment.index.items.packages.removePackageTypeYesNo.heading = Are you sure you want to remove this package from item {1} in house consignment {0}?
```
Comment (jackheslop96) [inline conf/messages.en:947]: Any reason we have {1} then {0} and don't just swap the args round?

Comment (justega247) [reply conf/messages.en:947]: Just following the pattern of specifying the houseConsignment index before the itemIndex @jackheslop96

File: app/controllers/houseConsignment/index/items/packages/RemovePackageTypeYesNoController.scala [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+  def onPageLoad(arrivalId: ArrivalId, houseConsignmentIndex: Index, itemIndex: Index, packageIndex: Index, mode: Mode): Action[AnyContent] =
+    actions
+      .getStatus(arrivalId)
+      .andThen(getMandatoryPage(PackageTypePage(houseConsignmentIndex, itemIndex, packageIndex))) {
+        implicit request =>
+          val quantity = request.userAnswers.get(NumberOfPackagesPage(houseConsignmentIndex, itemIndex, packageIndex)).map(_.toString())
+          val insetText: Option[String] = quantity.map(
+            numberOfPackages => s"$numberOfPackages ${packageType.toString}"
+          )
```
Comment (jackheslop96) [inline app/controllers/houseConsignment/index/items/packages/RemovePackageTypeYesNoController.scala:66]: Maybe a question for Gavin. What should the content be when only the package type is defined?

File: app/controllers/houseConsignment/index/items/packages/RemovePackageTypeYesNoController.scala [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+                } yield Redirect(addAnother(arrivalId, mode))
+            )
+      }
+
+  private def insetText(quantity: Option[String], packageType: PackageType): Option[String] =
+    quantity
+      .map(
+        value => s"$value ${packageType.toString}"
+      )
+      .orElse(Some(packageType.toString))
```
Comment (jackheslop96) [inline app/controllers/houseConsignment/index/items/packages/RemovePackageTypeYesNoController.scala:109]: `.orElse(Some(packageType.toString))` = `.getOrElse(packageType.toString)`

Comment (justega247) [reply app/controllers/houseConsignment/index/items/packages/RemovePackageTypeYesNoController.scala:109]: We use the Optionality in the View additionalHtml property, hence why I have gone with the above approach @jackheslop96 .

**Review by jackheslop96 — APPROVED** [review]

File: app/controllers/houseConsignment/index/items/packages/RemovePackageTypeYesNoController.scala (hunk 4 above)
Comment (justega247) [reply app/controllers/houseConsignment/index/items/packages/RemovePackageTypeYesNoController.scala:109]: I will look at moving the optionality logic to the view instead.

**Pull request merged**